src/Traits/InertiaResponseTrait.php

```diff
@@ -30,10 +33,40 @@ trait InertiaResponseTrait
         'JsonViewClass' => \CakeDC\Inertia\View\InertiaJsonView::class,
     ];
 
-    public function buildPaginationLinks(PaginatedInterface $paging)
+    public function parseRelated(array $data, string $model): array
+    {
+        foreach ($data as $key => $val) {
+            if (is_array($val)) {
+                if (array_key_exists('_ids', $val)) {
+                    $newItems = [];
+                    foreach ($val as $item) {
+                        foreach ($item as $value) {
+                            $newItems[] = $value['id'];
+                        }
+                    }
+                    $data[$key]['_ids'] = $newItems;
+                }
+            }
+        }
+
+        $db = ConnectionManager::get('default');
```

Comment:
what if the related table is in another datasource?

Reply:
could it be true, I pass the datasource as a parameter? then in parseRelated function we would have 
- data
- model
- source